### PR TITLE
Fix captchasrv potential problem with malformed input

### PR DESCRIPTION
**Pull request by fredericmora**
captchasrv: Fix a potential problem with malformed input

The captcha server did not correctly handle an empty input. It also accepted unreasonably long input.

### CODE REVIEW

**Comment by mikedlr** [comment]
Made this a draft - Fred had some second thoughts and is going to change some things.

**Comment by mikedlr** [comment]
When finished this should address https://github.com/spamgourmet/spamgourmet/issues/25

File: captchasrv/captchasrv.pl [hunk 1]
```diff
@@ -50,8 +50,10 @@
 ##########
 # Constant section - Personnalize here
 # Beware: constant module means don't put a $ in front of name!
-## Installation dir of this program - IMPORTANT
-use constant InstallDir => "/home/mora/src/spamgourmet/captcha";
+
+# Used to find data files in same dir as this script
+use constant InstallDir => dirname(__FILE__);
```
Comment (jgonggrijp) [inline captchasrv/captchasrv.pl:55]: These exact lines appear in more than one place,

File: captchasrv/mkquizword.pl [hunk 2]
```diff
@@ -22,13 +22,14 @@
...
+use File::Basename;
 
 sub mkquizworddict {
 
   ##########
   # Constant section - Personnalize here
-  ## Installation dir of this program - IMPORTANT
-  *InstallDir=\ "/home/mora/src/spamgourmet/captcha";
+  ## Installation dir of this program - Used to find data files in same dir as this scrupt
+  *InstallDir=\ dirname(__FILE__);;
```
Comment (jgonggrijp) [inline captchasrv/mkquizword.pl:32]: i.e., also here. Maybe factor out `InstallDir` to a common module?

File: captchasrv/captchasrv.pl [hunk 3]
```diff
@@ -147,27 +149,31 @@
...
+        # Send base name of captcha file to client
+        $bname = basename($fname);
+        print $client "$bname";
+        close $fh;
+        PrintTrace(3, "sent name \"$bname\" to client\n");
+    }
     close $client;
-    close $fh;
-    PrintTrace(3, "sent name \"$bname\" to client, closed client connection\n");
+     PrintTrace(2, "Closed client connection\n");
```
Comment (jgonggrijp) [inline captchasrv/captchasrv.pl:176]: You gained a leading space here.

**Review by jgonggrijp — COMMENTED** [review]
I have zero perl experience, so please take my review with a grain of salt.

File: captchasrv/mkquizword.pl (hunk 2 above)
Comment (fredericmora) [reply captchasrv/mkquizword.pl:32]: You don't want this symbol to go to a different file!

Comment (jgonggrijp) [reply captchasrv/mkquizword.pl:32]: Not even if it is in the same directory?

**Pull request merged**